pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTConstructorDeclaration.java

```diff
@@ -40,4 +41,25 @@ public boolean containsComment() {
     public void setContainsComment() {
         this.containsComment = true;
     }
+
+    @Override
+    public QualifiedName getQualifiedName() {
+        if (qualifiedName != null) {
+            return qualifiedName;
+        }
+
+        ASTClassOrInterfaceDeclaration parent = this.getFirstParentOfType(ASTClassOrInterfaceDeclaration.class);
+
+        ASTFormalParameters params = this.getFirstDescendantOfType(ASTFormalParameters.class);
+        int numParams = params.jjtGetNumChildren();
+        String[] types = new String[numParams];
+
+        for (int i = 0; i < numParams; i++) {
+            types[i] = params.jjtGetChild(i).getFirstDescendantOfType(ASTType.class).getTypeImage();
```

Comment:
I just verified with the Designer (because I was not sure, whether there would be always a Type node, even for primitive type - but yes, there is): the typeImage might not be sufficient in case of arrays, e.g.

    Foo(int a, int[] b)

Both typeImages for a and b are `int` (which might also be actually a bug in the existing PMD code), but `isArray()` is false for a and true for b.
In this example, the qualified name for another constructor

    Foo(int a, int b)

would be the same.

Just keep that in mind, when you write unit tests for this part.